test/steps/suite_steps.go

```diff
@@ -65,3 +74,35 @@ func scenarioName(args interface{}) string {
 	}
 	panic("unknown scenario type")
 }
+
+// hasFeatureTag indicates whether the given feature has a tag with the given name.
+func hasFeatureTag(feature *gherkin.Feature, name string) bool {
+	for _, tag := range feature.Tags {
+		if tag.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+// hasScenarioTag indicates whether the given scenario or scenario outline has a tag of the given name.
+func hasScenarioTag(args interface{}, name string) bool {
+	for _, tag := range scenarioTags(args) {
+		if tag.Name == name {
+			return true
+		}
+	}
+	return false
+}
+
+func scenarioTags(args interface{}) []*gherkin.Tag {
+	scenario, ok := args.(*gherkin.Scenario)
+	if ok {
+		return scenario.Tags
+	}
+	scenarioOutline, ok := args.(*gherkin.ScenarioOutline)
+	if ok {
+		return scenarioOutline.Tags
+	}
+	panic("unknown scenario type")
```

Comment:
Can we use a type switch here: https://tour.golang.org/methods/16

Reply:
That's a pretty cool trick! Definitely makes the code look cleaner.